File: Algorithms/Bit Manipulation/Maximizing XOR.py
import os

#
# Complete the 'maximizingXor' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER l
#  2. INTEGER r
#

def maximizingXor(l, r):
    # Write your code here
    max_xor = 0
    for i in range(l, r + 1):
        for j in range(i, r + 1):
            xor = i ^ j
            max_xor = max(max_xor, xor)
    return max_xor


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    l = int(input().strip())
    r = int(input().strip())

    result = maximizingXor(l, r)

    fptr.write(str(result) + '\n')
    fptr.close()

File: Algorithms/Bit Manipulation/Maximizing XOR.php
<?php

/*
 * Complete the 'maximizingXor' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER l
 *  2. INTEGER r
 */

function maximizingXor($l, $r) {
    # Write your code here
    $max = 0;
    for ($i = $l; $i <= $r; $i++) {
        for ($j = $l; $j <= $r; $j++) {
            $max = max($max, $i ^ $j);
        }
    }
    return $max;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$l = intval(trim(fgets(STDIN)));
$r = intval(trim(fgets(STDIN)));

$result = maximizingXor($l, $r);

fwrite($fptr, $result . "\n");
fclose($fptr);
